### PR TITLE
Model Evaluation component upgrade - Bug fixes, logging improvement and remove model uri

### PR DESCRIPTION
**Pull request by sarthaks95**

### CODE REVIEW

**Review by s-aditi — APPROVED** [review]

**Review by ankushbhatia2 — APPROVED** [review]

File: assets/training/model_evaluation/components/compute_metrics/spec.yaml [hunk 1]
```diff
@@ -69,16 +74,16 @@ outputs:
...
+  --task '${{inputs.task}}'
   --ground_truths '${{inputs.ground_truth}}'
   --predictions '${{inputs.prediction}}'
   --output '${{outputs.evaluation_result}}'
   $[[--prediction_probabilities '${{inputs.prediction_probabilities}}']]
   $[[--config-file-name '${{inputs.evaluation_config}}']]
-  $[[--ground_truths_column_name ${{inputs.ground_truth_column_name}}]]
-  $[[--predictions_column_name ${{inputs.prediction_column_name}}]]
+  $[[--ground_truths_column_name '${{inputs.ground_truth_column_name}}']]
+  $[[--predictions_column_name '${{inputs.prediction_column_name}}']]
```
Comment (ankushbhatia2) [inline assets/training/model_evaluation/components/compute_metrics/spec.yaml:88]: The single quotes are required to handle whitespaces

File: assets/training/model_evaluation/components/evaluate_model/spec.yaml [hunk 2]
```diff
@@ -80,21 +77,19 @@ outputs:
...
   python download_dependencies.py
-  --mlflow-model '${{inputs.mlflow_model}}'
-  $[[--model-uri ${{inputs.model_uri}}]] ;
+  --mlflow-model '${{inputs.mlflow_model}}' ;
   python evaluate_model.py
-  --task ${{inputs.task}}
+  --task '${{inputs.task}}'
   --output '${{outputs.evaluation_result}}'
-  --label-column-name ${{inputs.test_data_label_column_name}}
+  --label-column-name '${{inputs.test_data_label_column_name}}'
```
Comment (ankushbhatia2) [inline assets/training/model_evaluation/components/evaluate_model/spec.yaml:88]: Keep single quotes

File: assets/training/model_evaluation/components/model_prediction/spec.yaml [hunk 3]
```diff
@@ -90,26 +86,24 @@ outputs:
...
   --prediction-probabilities '${{outputs.prediction_probabilities}}'
   --ground-truth '${{outputs.ground_truth}}'
   --performance-metadata '${{outputs.performance_metadata}}'
   --mlflow-model '${{inputs.mlflow_model}}'
-  --device ${{inputs.device}}
+  --device '${{inputs.device}}'
   $[[--input-column-names '${{inputs.input_column_names}}']]
-  $[[--model-uri ${{inputs.model_uri}}]]
-  $[[--label-column-name ${{inputs.label_column_name}}]]
+  $[[--label-column-name '${{inputs.label_column_name}}']]
```
Comment (ankushbhatia2) [inline assets/training/model_evaluation/components/model_prediction/spec.yaml:104]: Keep single quotes

File: assets/training/model_evaluation/components/compute_metrics/spec.yaml (hunk 1 above)
Comment (sarthaks95) [reply assets/training/model_evaluation/components/compute_metrics/spec.yaml:88]: I've added the single quotes, not removed them.

File: assets/training/model_evaluation/components/model_prediction/spec.yaml (hunk 3 above)
Comment (sarthaks95) [reply assets/training/model_evaluation/components/model_prediction/spec.yaml:104]: I've added the single quotes, not removed them.

**Review by ankushbhatia2 — APPROVED** [review]

**Pull request merged**